manifests/certs.pp

```diff
@@ -1,17 +1,23 @@
-# @summary This class creates a certificates for Grafana and for connecting to PE Postgres.
+# @summary Copy Puppet Agent keypair for use by metric services.
 #
-# This class creates a set of certificates in /etc/${service}. These certificates
-# are used when configuring Grafana to use SSL and to connect to PE Postgres.
-# The certificates are based on the agent's own Puppet certificates.
+# This type creates copies of the Puppet Agent's SSL keypair in `/etc/${service}`
+# with user+group ownership set to `${service}`. These certificates are used
+# when configuring Grafana to use SSL and to connect Telegraf with PE Services.
 #
 # @param service
-#   The service name associated with these certificates.
+#   The service name to associate with the keypair copy.
 #
-define puppet_metrics_dashboard::certs(
-  $service = $name
+# @param ssl_dir
+#   The directory to copy Puppet Agent SSL files from. Defaults to the
+#   value of `puppet config print --section server ssldir` used by the
+#   Puppet Server, often `/etc/puppetlabs/puppet/ssl`. Use Hiera to
+#   override this value if agents have a different `ssldir` setting
+#   or if `bolt apply` is being used.
+define puppet_metrics_dashboard::certs (
+  $service = $name,
+  $ssl_dir = lookup('puppet_metrics_dashboard::certs::ssl_dir',
+                    {default_value => $settings::ssldir}),
```

Comment:
This default of `$settings::ssldir` preserves backwards compatibility, but I'd argue `/etc/puppetlabs/puppet/ssl` would be a better default as:

  - It would make the module usable with `bolt apply` without configuration.
  - `$settings::ssldir` gives you the SSL directory of the _Puppet Server_ which is in no way guaranteed to be relevant to the Puppet Agent applying this configuration. If folks have customized the SSL directory to be something other than the default, it is probably less confusing for everyone to make them straight-up configure that via Hiera.

Reply:
+1 on changing to `/etc/puppetlabs/puppet/ssl`. Especially since `/etc/puppetlabs/puppetserver/ssl` will become more popular in 7.x/2021.x.

Reply:
I can change it to `/etc/puppetlabs/puppet/ssl` if the next release will be 3.0.

I think the Puppet 7 change to `/etc/puppetlabs/puppetserver/ssl` was for the `cadir` setting, not `ssldir`.

Reply:
I'll use `/etc/puppetlabs/puppet/ssl` for 3.0, I think this is good for now :+1: